catchup/service.go

```diff
@@ -494,6 +509,18 @@ func (s *Service) pipelinedFetch(seedLookback uint64) {
 				return
 			}
 
+			fetchTime := time.Now()
+			fetchDur := fetchTime.Sub(s.prevBlockFetchTime)
```

Comment:
on the first iteration `s.prevBlockFetchTime` is zero => fetchDur is `time.Now()` so that always rate limited causing slow catchup at start

Reply:
Correct. The first 2 blocks would be downloaded before ramping up the parallelization. This is a fixed overhead while the service is starting.